### PR TITLE
Fix some lints documentation

### PR DESCRIPTION
**Pull request by hafeoz**
Several lints documentation failed to show the output of the example (mostly due to `ignore` attribute):

- [irrefutable_let_patterns](https://doc.rust-lang.org/rustc/lints/listing/warn-by-default.html#irrefutable-let-patterns)
- [asm_sub_register](https://doc.rust-lang.org/rustc/lints/listing/warn-by-default.html#asm-sub-register)
- [bad_asm_style](https://doc.rust-lang.org/rustc/lints/listing/warn-by-default.html#bad-asm-style)
- [ineffective_unstable_trait_impl](https://doc.rust-lang.org/rustc/lints/listing/deny-by-default.html#ineffective-unstable-trait-impl)
- duplicate_macro_attributes

This pull request fixes these lints output so that they can be displayed properly.

### CODE REVIEW

**Comment by hafeoz** [comment]
`asm_sub_register` and `bad_asm_style` does work in above's link. I'm not sure why it's not working in local environment before the PR. `irrefutable_let_patterns` and `ineffective_unstable_trait_impl` is indeed broken in above's link.

File: compiler/rustc_lint/src/builtin.rs [hunk 1]
```diff
@@ -3147,6 +3147,7 @@ declare_lint! {
     /// ### Example
     ///
     /// ```rust,compile_fail
+    /// # #![feature(asm)]
```
Comment (ehuss) [inline compiler/rustc_lint/src/builtin.rs:3150]: This shouldn't be necessary.  The asm feature was stabilized in the 1.59 release.

**Review by ehuss — COMMENTED** [review]
Thanks for the fixups!

File: compiler/rustc_lint/src/builtin.rs (hunk 1 above)
Comment (hafeoz) [reply compiler/rustc_lint/src/builtin.rs:3150]: Got it!

**Comment by ehuss** [comment]
Thanks!

r? @ehuss

@bors r+ rollup

**Pull request merged**